eip/src/test/java/akka/stream/alpakka/eip/javadsl/SplitterExamples.java

```diff
@@ -0,0 +1,80 @@
+package akka.stream.alpakka.eip.javadsl;
+
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+import akka.testkit.JavaTestKit;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+
+public class SplitterExamples {
+    private static ActorSystem system;
+    private static Materializer materializer;
+
+    @Test
+    public void simpleSplit() throws ExecutionException, InterruptedException {
+        //#Simple-Split
+        //Sample Source
+        Source<String, NotUsed> source = Source.from(Arrays.asList(new String[]{"1-2-3", "2-3", "3-4"}));
+
+
+        CompletionStage<List<Integer>> ret =
+                source.map(s -> Arrays.asList(s.split("-")))
+                        .mapConcat(f -> f)
+                        .map(s -> Integer.valueOf(s))
+                        .runWith(Sink.seq(), materializer);
+
+        //Verify results
+        List<Integer> list = ret.toCompletableFuture().get();
+        assert list.equals(Arrays.asList(1, 2, 3, 2, 3, 3, 4));
+        //#Simple-Split
+    }
+
+
+    @Test
+    public void splitAggregate() throws ExecutionException, InterruptedException {
+        //#Aggregate-Split
+        //Sample Source
+        Source<String, NotUsed> source = Source.from(Arrays.asList(new String[]{"1-2-3", "2-3", "3-4"}));
+
+
+        CompletionStage<List<Integer>> ret =
+                source.map(s -> Arrays.asList(s.split("-")))
+                        //split all messages into sub-streams
+                        .splitWhen(a -> true)
```

Comment:
Note that this has a high overhead because of the materialization of a substream for each element. Unless the individual aggregates are really large or the processing done for the sub stream is very cpu intense just materialising will likely dominate execution. 

In many if not most cases just putting the split-process-aggregate of each element in a single map. I wonder if it wouldn't be better to do that, with a `mapAsync` to achieve some parallelism in processing instead.

Reply:
Hmm interesting point, I never looked from this perspective.  My real life examples for using this pattern almost always involved some serious processing in the sub-streams. Either some heavy string parsing or I/O. I/O in sub-process  is actually good case for this pattern as usually involves more parallelism in the sub-process which adds the need of the back-pressure. (I so wished Camel has some elegant solution for this ... but that's why I am here :) )

Adding parallelism in the sub-stream is something that I think is out of scope of this pattern. 

Of course if your data is small and you just split for a very simple aggregation (like I did in the example) this pattern is actually overkill. You better just put your code in a map (sync or async) as you suggested and that's it. But using this pattern only for this is a little bit of going over yourself :). My usage intention was mostly on the examples that I described above (and that is how I use it in real life).

Reply:
Fair enough, maybe make sure that it is clear from the accompanying text that this is just for example and when it would make sense in real life?

Reply:
reworked the texts